src/lib/reasoners/sat_solver_sig.ml

```diff
@@ -39,6 +39,7 @@ type timeout_reason =
 type unknown_reason =
   | Incomplete
   | Memout
+  | Steps_limit of int
```

Comment:
`Step_limit`*

Reply:
@hra687261's remark still applies.

Reply:
This is just a copy of the previous ADT, which uses plural for steps. I may not be as fluent as you are in English, but my guess is "step limit" is okay to mean "this step is the limit" and "steps limit" is also ok to mean "this number of steps is the limit".

Reply:
It doesn't really matter (and I wouldn't have said anything if @hra687261 did not make a remark), but I'm pretty sure in this situation "steps limit" is incorrect, or at least not idiomatic (see for instance [here](https://english.stackexchange.com/questions/229406/how-to-correctly-write-chains-of-nouns-with-plurals-like-messages-list-and-ap)). It is a "resource limit", a "character limit", and a "step limit". This is different from French where we would say "la limite des pas", and "la limite du pas" would be wrong (I guess we could say "the limit of the steps" but it would be weird).

Reply:
Just noted I defined the exception Step_limit_reached :sweat_smile: